core/tests/tests/Provider/AnimationFrame.mint

```diff
@@ -1,23 +1,30 @@
 component Test.Provider.AnimationFrame {
-  state frames : Number = 0
+  state timestamp : Number = 0
 
-  use Provider.AnimationFrame { frames = () : Promise(Never, Void) { next { frames = frames + 1 } } }
+  use Provider.AnimationFrame {
+    frames =
+      (timestamp : Number) : Promise(Never, Void) {
+        next { timestamp = timestamp }
+      }
+  }
 
   fun render : Html {
     <div>
-      <{ Number.toString(frames) }>
+      <{ Number.toString(timestamp) }>
     </div>
   }
 }
 
 suite "Provider.AnimationFrame.frames" {
   test "called on an animation frame" {
-    with Test.Html {
-      <Test.Provider.AnimationFrame/>
-      |> start()
-      |> assertTextOf("div", "1")
-      |> assertTextOf("div", "2")
-      |> assertTextOf("div", "3")
+    with Test.Context {
+      with Test.Html {
+        <Test.Provider.AnimationFrame/>
+          |> start()
+          |> find("div")
+          |> map(Dom.getTextContent)
+          |> assertNotEqual("0")
```

Comment:
We shouldn't need `assertNotEqual `:

```suggestion
        (<Test.Provider.AnimationFrame/>
          |> start()
          |> find("div")
          |> map(Dom.getTextContent)) != "0"
```

Reply:
Btw, I don't think we even need to test the argument in any way ;)
We ain't responsible for its lifecycle, nor validity, so... 🤷‍♂️

Reply:
This wouldn't work since the left side is a `Test.Context(String)` and the right side is a `String`.

How tests work is that either it returns:

* `Bool` - if its `true` the test passes and if it's `false` the tests fails
* `Test.Context(a)` - a multi-step test, the steps are basically promises if step (promise) fails then the whole test fails, if all resolves then the test passes.